optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/constraint/ConstraintMatchJacksonJsonDeserializer.java

```diff
@@ -0,0 +1,43 @@
+package org.optaplanner.persistence.jackson.api.score.constraint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.persistence.jackson.api.score.DeserializationUtils;
+
+public class ConstraintMatchJacksonJsonDeserializer extends JsonDeserializer<ConstraintMatch> {
+
+    @Override
+    public ConstraintMatch deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        try {
+            JsonNode tree = parser.getCodec().readTree(parser);
+
+            String constraintPackage = tree.get("constraintPackage").asText();
+            String constraintName = tree.get("constraintName").asText();
+
+            Score<?> score = DeserializationUtils.deserializePolymorphicProperty(context, tree, "score");
+
+            List<Object> justificationList = new ArrayList<>();
+            tree.get("justificationList").forEach((t) -> {
+                try {
+                    justificationList.add(DeserializationUtils.deserializePolymorphicProperty(context, t,
+                            "justification"));
+                } catch (Exception e) {
```

Comment:
We never catch Exception or RuntimeException (unless there is a very very very good reason to do so)